app/services/membership_modifier.rb

```diff
@@ -0,0 +1,38 @@
+class MembershipModifier
+  class ExistingMembership < StandardError; end
+  class NoExistingMembership < StandardError; end
+
+  def self.create_membership(requesting_user:, joinable:, target_user:, admin:)
+    return unless joinable.admin? requesting_user
+
+    if joinable.contains? target_user
+      fail MembershipModifier::ExistingMembership
+    end
+
+    if admin
+      Membership.create(user: target_user, joinable: joinable).admin!
+    else
+      Membership.create(user: target_user, joinable: joinable).member!
+    end
```

Comment:
Rather than passing `true/false` for admin, could we pass the role, and then not have to use the conditional?:
```
Membership.create(user: target_user, joinable: joinable).send("#{role}!")
```
Might be better if we add more roles in the future?